tests/lib/rules/jsx-no-useless-fragment.js

```diff
@@ -0,0 +1,172 @@
+    },
+    '<NotFragment />',
+    '<React.NotFragment />',
+    '<NotReact.Fragment />',
+    {
+      code: '<Foo><><div /><div /></></Foo>',
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<div p={<>{"a"}{"b"}</>} />',
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<Fragment key={item.id}>{item.value}</Fragment>',
+      parser: parsers.BABEL_ESLINT
+    }
+  ],
+  invalid: [
+    {
+      code: '<></>',
+      output: null,
+      errors: [{messageId: 'NeedsMoreChidren', type: 'JSXFragment'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<>foo</>',
+      output: null,
+      errors: [{messageId: 'NeedsMoreChidren'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<><div/></>',
+      output: null,
+      errors: [{messageId: 'NeedsMoreChidren'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<Fragment />',
+      errors: [{messageId: 'NeedsMoreChidren'}]
+    },
+    {
+      code: `
+        <React.Fragment>
+          <Foo />
+        </React.Fragment>
+      `,
+      errors: [{messageId: 'NeedsMoreChidren'}]
+    },
+    {
+      code: `
+        <SomeReact.SomeFragment>
+          {foo}
+        </SomeReact.SomeFragment>
+      `,
+      settings: {
+        react: {
+          pragma: 'SomeReact',
+          fragment: 'SomeFragment'
+        }
+      },
+      errors: [{messageId: 'NeedsMoreChidren'}]
+    },
+    {
+      code: '<></>',
+      errors: [{messageId: 'NeedsMoreChidren'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<Eeee><>foo</></Eeee>',
+      output: '<Eeee>foo</Eeee>',
+      errors: [{messageId: 'NeedsMoreChidren'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<div><>foo</></div>',
+      output: '<div>foo</div>',
+      errors: [{messageId: 'NeedsMoreChidren'}, {messageId: 'ChildOfHtmlElement'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<div><>{"a"}{"b"}</></div>',
+      output: '<div>{"a"}{"b"}</div>',
+      errors: [{messageId: 'ChildOfHtmlElement'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: `
+        <section>
+          <Eeee />
+          <Eeee />
+          <>{"a"}{"b"}</>
+        </section>`,
+      output: `
+        <section>
+          <Eeee />
+          <Eeee />
+          {"a"}{"b"}
+        </section>`,
+      errors: [{messageId: 'ChildOfHtmlElement'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<div><Fragment>{"a"}{"b"}</Fragment></div>',
+      output: '<div>{"a"}{"b"}</div>',
+      errors: [{messageId: 'ChildOfHtmlElement'}]
+    },
+    {
+      code: `
+        // not safe to fix
+        <section>
+          git<>
+            <b>hub</b>.
+          </>
+        </section>`,
+      output: null,
+      errors: [{messageId: 'ChildOfHtmlElement'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<div>a <>{""}{""}</> a</div>',
+      output: null,
```

Comment:
this one seems autofixable, since it's the child of an html element?

Reply:
Now fixed.